initial implementation of npipe

jaytaph opened the pull request:
Implementation of the npipe transport. The transport code itself can be used on unix as well (it is defined as a valid transport, just like the unixSocketTransport is on windows platforms), but the actual dialer is windows only.

The unix tests do not run correctly on windows, and vice versa, so i've made sure only the correct tests are run on the correct builds.

cpuguy83 commented on transport/npipe.go:
```diff
@@ -0,0 +1,43 @@
+)
+
+func DefaultWindowsTransport() *Transport {
+	t, _ := NpipeTransport("//./pipe/docker_engine")
```
I'm not sure why I ignored errors on the unix socket version of this.
Seems like a mistake.

Can we propagate the error? Bonus points if you want to fix the unix one too 👼

jaytaph commented:
I've added the changes (although it's not propagated all the way down to NewClient, we could change that too though)..

I also added a "DefaultTransport" which is OS independent so it might be easier to fetch the correct transport for clients.

cpuguy83 reviewed (approved):
LGTM

Thanks!
